llama_index/vector_stores/cassandra.py

```diff
@@ -0,0 +1,302 @@
+            yield this_batch
+            this_batch = []
+    if this_batch:
+        yield this_batch
+
+
+class CassandraVectorStore(VectorStore):
+    """Cassandra Vector Store.
+
+    An abstraction of a Cassandra table with
+    vector-similarity-search. Documents, and their embeddings, are stored
+    in a Cassandra table and a vector-capable index is used for searches.
+    The table does not need to exist beforehand: if necessary it will
+    be created behind the scenes.
+
+    All Cassandra operations are done through the cassIO library.
+
+    Args:
+        session (cassandra.cluster.Session): the Cassandra session to use
+        keyspace (str): name of the Cassandra keyspace to work in
+        table (str): table name to use. If not existing, it will be created.
+        embedding_dimension (int): length of the embedding vectors in use.
+        ttl_seconds (Optional[int]): expiration time for inserted entries.
+            Default is no expiration.
+
+    """
+
+    stores_text: bool = True
+    flat_metadata: bool = True
+
+    def __init__(
+        self,
+        session: Any,
+        keyspace: str,
+        table: str,
+        embedding_dimension: int,
+        ttl_seconds: Optional[int] = None,
+        insertion_batch_size: int = DEFAULT_INSERTION_BATCH_SIZE,
+    ) -> None:
+        import_err_msg = "`cassio` package not found, please run `pip install cassio`"
+        try:
+            from cassio.table import ClusteredMetadataVectorCassandraTable  # noqa: F401
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        self._session = session
+        self._keyspace = keyspace
+        self._table = table
+        self._embedding_dimension = embedding_dimension
+        self._ttl_seconds = ttl_seconds
+        self._insertion_batch_size = insertion_batch_size
+
+        _logger.debug("Creating the Cassandra table")
+        self.vector_table = ClusteredMetadataVectorCassandraTable(
+            session=self._session,
+            keyspace=self._keyspace,
+            table=self._table,
+            vector_dimension=self._embedding_dimension,
+            primary_key_type=["TEXT", "TEXT"],
+            # a conservative choice here, to make everything searchable
+            # except the bulky "_node_content" key (it'd make little sense to):
+            metadata_indexing=("default_to_searchable", ["_node_content"]),
+        )
+
+    def add(
+        self,
+        embedding_results: List[NodeWithEmbedding],
+    ) -> List[str]:
+        """Add embedding results to index.
+
+        Args
+            embedding_results: List[NodeWithEmbedding]: list of embedding results
+
+        """
+        node_ids = []
+        node_contents = []
+        node_metadatas = []
+        node_embeddings = []
+        for result in embedding_results:
+            metadata = node_to_metadata_dict(
+                result.node,
+                remove_text=False,  # must keep the text here for retrieval
```

Comment:
High level comment -- why can't we remove the text here? Right now, the text is being stored twice actually. Once in the `_node_content` field and again in the `node_contents` field.

Is the `node_contents` field not available in the query result? Normally you'd parse that out and do something like `node.set_content(node_text)`

Reply:
Thank you, well spotted! I adapted the code accordingly and tested all my example codes using it. This should now be all right.